spanner/value.go

```diff
@@ -1877,6 +1969,33 @@ func decodeValue(v *proto3.Value, t *sppb.Type, ptr interface{}, opts ...decodeO
 			return errBadEncoding(v, err)
 		}
 		reflect.ValueOf(p).Elem().SetInt(y)
+	case *NullEnum:
+		if p == nil {
+			return errNilDst(p)
+		}
+		if p.EnumVal == nil {
+			return errNilDst(p.EnumVal)
+		}
+		if reflect.ValueOf(p.EnumVal).Kind() != reflect.Ptr {
+			return errNotAPointer(p)
+		}
+		if code != sppb.TypeCode_ENUM && code != sppb.TypeCode_INT64 {
+			return errTypeMismatch(code, acode, ptr)
+		}
+		if isNull {
+			*p = NullEnum{}
+			break
+		}
+		x, err := getStringValue(v)
```

Comment:
Line 1989 to 1996 is repeated multiple times. Should we extract method for them?

Reply:
I have created a method for the same.